privaterelay/management/commands/aggregate_generated_for.py

```diff
@@ -0,0 +1,129 @@
+from argparse import ArgumentParser
+from csv import DictReader, DictWriter
+from pathlib import Path
+from typing import Any
+from urllib.parse import urlparse
+
+from django.core.management.base import BaseCommand, CommandError
+
+
+def normalize(url: str) -> str:
+    """
+    The url in data may not have // which urlparse requires to recognize the netloc
+    script from:
+    https://stackoverflow.com/questions/53816559/python-3-netloc-value-in-urllib-parse-is-empty-if-url-doesnt-have
+    """
+    if not (
+        url.startswith("//") or url.startswith("http://") or url.startswith("https://")
+    ):
+        return "//" + url
+    return url
+
+
+def aggregate_by_generated_for(file_path: str) -> Path:
```

Comment:
Nit: I'd split this into three parts:

1. Load data from CSV into a `dict`
2. Aggregate data from a `dict` into a smaller `dict`
3. Save aggregated data to CSV

This way, in the future the management command could query the data directly, and feed the data to the second function, and nothing else changes. The middle function would be data-only, so it is more flexible about input and output and avoids creating files for testing.

Reply:
> * Load data from CSV into a `dict`

Moved this to the command handler

> * Aggregate data from a `dict` into a smaller `dict`

Simplified the `aggregate_by_generated_for` to only aggregate and return the smaller aggregate dictionary

> * Save aggregated data to CSV

Moved the logic from `aggregate_by_generated_for` and created `geneate_csv_file` to generate the aggregate dictionary to a CSV file